### DISCUSSION TITLE
Functions is returning Internal Server Error

### DISCUSSION THREAD

**Discussion by rbkayz** [supabase/supabase#7301] [Questions] [answered]
Folks,

I'm trying to run the basic hello-world function from Powershell

curl.exe -L -X POST 'https://welladjldntnorgqjmvm.functions.supabase.co/hello-world' -H 'Authorization: Bearer [my anon token]' --data '{"name":"Functions"}'

However receiving a 500 Internal Server Error. Here's the invocation log from the studio

Thank you for your help!

> Status
500
Method
POST
Timestamp
16 Jun, 2022 14:23
Execution Time
610ms
Deployment ID
welladjldntnorgqjmvm_10c7a06e-a808-4532-bfe3-db189f56f2bf_1
Log ID
86df50af-bd11-419e-98a4-3701a354f1e5
Request Path
/hello-world
Request body
{
  "headers": [
    {
      "accept": "*/*",
      "accept_encoding": null,
      "connection": null,
      "content_length": "16",
      "cookie": null,
      "host": "welladjldntnorgqjmvm.functions.supabase.co",
      "user_agent": "curl/7.79.1"
    }
  ],
  "method": "POST",
  "pathname": "/hello-world",
  "protocol": "https:",
  "search": null
}
Response POST body
{
  "headers": [
    {
      "content_length": null,
      "content_type": "text/plain;charset=UTF-8",
      "date": "Thu, 16 Jun 2022 08:53:14 GMT",
      "server": "deno/asia-south1-c",
      "vary": "Accept-Encoding"
    }
  ],
  "status_code": 500
}

**Comment by GaryAustin1** [comment]
Have you checked the log for that function in the dashboard?

**Reply by rbkayz** [reply]
This is what is given there. How can I read this?

Severity
INFO
Log ID
a4fa3d30-d4d9-4b2d-88d6-4a9451ffd076
Deployment version
1
Timestamp
16 Jun, 2022 20:14
Execution ID
d882dfd9-f30c-4c32-af2b-2cd5ac830aac
Deployment ID
welladjldntnorgqjmvm_10c7a06e-a808-4532-bfe3-db189f56f2bf_1
Metadata
{
  "deployment_id": "welladjldntnorgqjmvm_10c7a06e-a808-4532-bfe3-db189f56f2bf_1",
  "event_type": "log",
  "execution_id": "d882dfd9-f30c-4c32-af2b-2cd5ac830aac",
  "function_id": "10c7a06e-a808-4532-bfe3-db189f56f2bf",
  "level": "info",
  "project_ref": "welladjldntnorgqjmvm",
  "timestamp": "2022-06-16T14:44:37.053080265+00:00",
  "version": "1"
}

**Reply by GaryAustin1** [reply]
Sorry, I saw 500 error and thought that you were just looking at the return from the call.  If that is all that is in the log then not very useful.

**Comment by thorwebdev** [comment, marked as answer]
@rbkayz can you try [running the function locally](https://supabase.com/docs/reference/cli/supabase-functions-serve) and check the terminal output? Also make sure to upgrade to the latest supabase cli version (`brew upgrade supabase`)

**Reply by rbkayz** [reply]
All good. Works well on postman and via js. Something is weird with running this via curl from terminal
